src/common/hooks/use-chain-mode.ts

```diff
@@ -4,41 +4,43 @@ import { useModal } from '@common/hooks/use-modal';
 import { useNetworkMode } from '@common/hooks/use-network-mode';
 import { IS_BROWSER } from '@common/constants';
 import { NetworkMode, NetworkModes } from '@common/types/network';
+import { useAuthState } from '@common/hooks/use-auth';
 
 type ChainMode = NetworkMode | undefined;
 type SetChainMode = (mode: ChainMode) => Promise<void>;
 
 // kind of like useState, but for the query param state
 export const useChainMode = (): [ChainMode, SetChainMode] => {
   const router = useRouter();
+  const { setAuthResponse } = useAuthState();
 
-  const authResponse = router.query?.authResponse
-    ? { authResponse: router.query?.authResponse }
-    : {};
+  const setChainMode = async (chain: ChainMode) => {
```

Comment:
How come this is no longer in a usecallback?

Reply:
It was causing some issues -- might have removed it while debugging

Reply:
I have a new pr that improves some of this, will make this change in that PR